common/Resources/ti.internal/extensions/node/buffer.js

```diff
@@ -1347,6 +1376,19 @@ class Buffer {
 				value.copy(buffer, 0, 0, length);
 				return buffer;
 			}
+			if (value instanceof Uint8Array) {
```

Comment:
combine with line 1361 above (the Array case) basically just have the if check for `Array.isArray(value) || value instanced Uint8Array` as both cases iterate through and copy the bytes. The masking with 0xFF should basically be a no-op for a Uint8Array's values.

Then we can gain the fix for empty array you did here.